### PR TITLE
Introducing type stubs for the dateparser module

### PR DESCRIPTION
**Pull request by xcthulhu**

### CODE REVIEW

File: third_party/2and3/dateparser.pyi [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import datetime
+from typing import Optional, List, Dict
+
+__version__ = str
```
Comment (JelleZijlstra) [inline third_party/2and3/dateparser.pyi]: Surely should be `__version__: str`?

File: third_party/2and3/dateparser.pyi [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+def parse(
+    date_string: str,
+    date_formats: Optional[List[str]] = None,
```
Comment (JelleZijlstra) [inline third_party/2and3/dateparser.pyi]: From the code it looks like these list arguments can also be tuples or sets; there is an isinstance check in `DateDataParser.__init__`.

Comment (xcthulhu) [reply third_party/2and3/dateparser.pyi]: I see this now.  The documentation says the arguments are lists which confused me.  I've tried to make this change in the latest revision (which I've force pushed).

**Comment by xcthulhu** [comment]
> A few other functions are documented in https://dateparser.readthedocs.io/en/latest/, so this stub should be marked as incomplete with a `__getattr__` function.

I've added

```
def __getattr__(name: str) -> Any: ...  # incomplete
```

Following the example I've found here: https://github.com/xcthulhu/typeshed/blob/1aba324d9b10aead89cf7884100a75beadaa90dd/third_party/2and3/cryptography/x509.pyi#L299

Is this right?

**Comment by JelleZijlstra** [comment]
This looks mostly OK now, but could you fix the CI errors?

**Comment by JelleZijlstra** [comment]
(And sorry for the delay in reviewing!)

**Comment by srittau** [comment]
Obsoleted by #4106.

**Pull request closed without merging**